cluster-autoscaler/core/scaledown/actuation/actuator.go

```diff
@@ -159,18 +160,33 @@ func (a *Actuator) deleteAsyncEmpty(NodeGroupViews []*budgets.NodeGroupView) (re
 func (a *Actuator) taintNodesSync(NodeGroupViews []*budgets.NodeGroupView) errors.AutoscalerError {
 	var taintedNodes []*apiv1.Node
 	for _, bucket := range NodeGroupViews {
+		startTimestamp := map[string]time.Time{}
```

Comment:
I think a lot of the code here could be moved to `UpdateLatencyTracker`. Specifically:
- `startTimestamp` is already a member of `UpdateLatencyTracker`, it actually breaks encapsulation to update it here. What's worse, you're writing it from one goroutine and reading from another without any synchronization.
- `resultChan` isn't actually necessary. You can have a synchronous interface:

    ```
    updateLatencyTracker := NewUpdateLatencyTracker(a.ctx.AutoscalingKubeClients.ListerRegistry.AllNodeLister())
    updateLatencyTracker.Start()
    (...)
    updateLatencyTracker.Add(node.Name, time.Now())
    (...)
    if err != nil { updateLatencyTracker.Stop() }
    (...)
    latency := updateLatencyTracker.Await()
    ```

- or async:

    ```
    updateLatencyTracker := NewUpdateLatencyTracker(a.ctx.AutoscalingKubeClients.ListerRegistry.AllNodeLister())
    go updateLatencyTracker.Start()
    (...)
    updateLatencyTracker.StartTimes <- startTime{node.Name, time.Now()}
    (...)
    if err != nil { close(updateLatencyTracker.StopChan) }
    (...)
    latency := <-updateLatencyTracker.ResultChan
    ```

Now it's mixed, which might cause bugs like when trying to update the map directly.

Reply:
Done